api/jobs/handlers.py

```diff
@@ -301,67 +259,51 @@ def stats(self):
 
         return Queue.get_statistics(tags=tags, last=last, unique=unique, all_flag=all_flag)
 
+    @require_admin
     def pending(self):
-        if not self.superuser_request and not self.user_is_admin:
-            self.abort(403, 'Request requires admin')
-
         tags = self.request.GET.getall('tags')
         if len(tags) == 1:
             tags = tags[0].split(',')
 
         return Queue.get_pending(tags=tags)
 
+    @require_admin
     def next(self):
-
-        if not self.superuser_request and not self.user_is_admin:
-            self.abort(403, 'Request requires admin')
-
-
         tags = self.request.GET.getall('tags')
         if len(tags) <= 0:
             tags = None
 
         job = Queue.start_job(tags=tags)
 
         if job is None:
-            self.abort(400, 'No jobs to process')
+            raise InputValidationException('No jobs to process')
```

Comment:
Since we're adjusting this anyway, could we use an `APINotFoundException`/`404`? The client didn't make an improperly formatted request.

Reply:
Uh, good question... The route exists and is valid, so a 404 does not feel appropriate here. I wanted to use something that kept the 400, hence my choice. I could make a new error class if you wanted, but it just seemed so trivial I didn't bother initially.

Reply:
404 as in a "next job" is missing. IMO an empty response would be the best way to respond but easier to write around a non-200. Not a huge deal we change it.

Reply:
Yeah, the trick here is I didn't want to change behavior for the existing clients. I could go check or change our consumers to not mind it, but it doesn't seem worth a breaking change.